File: Backend/Todo-Service/Todo-Service/Dtos/UserDto.cs
﻿using MongoDB.Bson;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Todo_Service.Dtos
{
	public class UserDto
	{
		public string Id { get; set; }
		public string FirstName { get; set; }
		public string LastName { get; set; }
		public string UserName { get; set; }
		public string Password { get; set; }
	}
}

File: Backend/todo-express-service/src/controllers/dtos/user-dto.ts
import { User } from "../../models/user";

export class UserDto {
    constructor(
        public id: string,
        public firstName: string,
        public lastName: string,
        public userName: string,
        public password: string) { }

    public static fromUser(user: User): UserDto {
        return new UserDto(user._id,
            user.firstName,
            user.lastName,
            user.userName,
            user.password)
    }
}